pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -195,29 +207,39 @@ impl<T: Config> Module<T> {
             public_ips_list: Vec::new(),
         };
 
+        // Create contract
+        let twin_id = pallet_tfgrid::TwinIdByAccountID::<T>::get(&account_id);
         let contract = Self::_create_contract(
+            id,
             twin_id,
             types::ContractData::NodeContract(node_contract.clone()),
         )?;
 
+        // Insert contract id by (node_id, hash)
         ContractIDByNodeIDAndHash::insert(node_id, deployment_hash, id);
 
+        // Insert contract into active contracts map
         let mut node_contracts = ActiveNodeContracts::get(&node_contract.node_id);
         node_contracts.push(id);
         ActiveNodeContracts::insert(&node_contract.node_id, &node_contracts);
 
+        // Insert resources for node contract
+        let node_contract_resources = types::ContractResources::default();
```

Comment:
In contract creation, why do you need to create a default "contract resources" object if the node going to set it up again later if it has been deployed?

Reply:
Because this function https://github.com/threefoldtech/tfchain/blob/9d213bbe0bc72a95fbb1df9f2c17ec3017e4b2d8/pallets/pallet-smart-contract/src/lib.rs#L645

which is executed every billing cycle gets the contract resources object and calculates the amount due. The code will know when these values are 0 it should return 0 cost but there is a chance a contract also has a public IP. And this function also calculates the cost for the public IP even though there are no contract resources reported

Reply:
indeed, we should not have this default object for every created contract. Not having this will limit storage :)